dags/analyzer_helper/discord/fetch_discord_platforms.py

```diff
@@ -0,0 +1,73 @@
+from hivemind_etl_helpers.src.utils.mongo import MongoSingleton
+
+
+class FetchDiscordPlatforms:
+    """
+    A class to fetch Discord platform data from a MongoDB collection.
+
+    Attributes:
+        client (MongoClient): The MongoDB client.
+        db (Database): The MongoDB database.
+        collection (Collection): The MongoDB collection.
+    """
+
+    def __init__(self, db_name="Core", collection="platforms"):
+        """
+        Initializes the FetchDiscordPlatforms class.
+
+        Args:
+            db_name (str): The name of the database. Defaults to 'Core'.
+            collection (str): The name of the collection. Defaults to 'platforms'.
+        """
+        self.client = MongoSingleton.get_instance().client
+        self.db = self.client[db_name]
+        self.collection = self.db[collection]
+
+    def fetch_all(self):
+        """
+        Fetches all Discord platforms from the MongoDB collection.
+
+        Returns:
+            list: A list of dictionaries, each containing platform data with the following fields:
+                - platform_id: The platform ID (_id from MongoDB).
+                - metadata: A dictionary containing action, window, period, selectedChannels, and id.
+                - recompute: A boolean set to False.
+        """
+        query = {
+            "disconnectedAt": None,
+            "platform": "discord",
+        }
+        # TODO: Understand if we need to expand projection with:
+        # isInProgress
+        # icon
+        # name
+        # analyzerStartedAt
+        projection = {
+            "_id": 1,
+            "metadata.action": 1,
+            "metadata.window": 1,
+            "metadata.period": 1,
+            "metadata.selectedChannels": 1,
+            "metadata.id": 1,
+        }
```

Comment:
We don't need the `metadata.action`, `metadata.window`, and `metadata.selectedChannels` for data transformation process. They are input parameters for analyzer, which we could fetch before calling the analyzer in another task.
We should make sure to pass the least amount of possible data between airflow tasks.